src/main/scala/mesosphere/marathon/api/v2/InfoEmbedResolver.scala

```diff
@@ -46,13 +46,30 @@ private[v2] object AppInfoEmbedResolver {
     }
 
     def separatePrefix(embedMe: String): (String, String) = {
-      val removablePrefix = EmbedAppsPrefixes.find(embedMe.startsWith(_)).getOrElse("")
+      val removablePrefix = EmbedAppsPrefixes.find(embedMe.startsWith).getOrElse("")
       val withoutPrefix = embedMe.substring(removablePrefix.length)
       (removablePrefix, withoutPrefix)
     }
 
-    import scala.collection.JavaConverters._
-    val embedWithSeparatedPrefixes = embed.asScala.map(separatePrefix)
-    embedWithSeparatedPrefixes.flatMap { case (prefix, withoutPrefix) => mapEmbedStrings(prefix, withoutPrefix) }.toSet
+    val embedWithSeparatedPrefixes = embed.map(separatePrefix)
+    embedWithSeparatedPrefixes.flatMap { case (prefix, withoutPrefix) => mapEmbedStrings(prefix, withoutPrefix) }
+  }
+
+  def resolveGroup(embeds: Set[String]): Set[GroupInfo.Embed] = {
+    embeds.flatMap {
+      case EmbedGroups => Some(GroupInfo.Embed.Groups)
+      case EmbedApps   => Some(GroupInfo.Embed.Apps)
+      case unknown: String =>
+        log.warn(s"unknown embed argument: $unknown")
```

Comment:
maybe use "unknown group embed argument: ..." here and "unknown app embed argument: ..." above so that we can distinguish the two log statements?